### PR TITLE
[SPARK-37966][SQL] Static partition insert should write _SUCCESS under partition location

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Currently, when we use static partition insert to write a partition with data source API, the `_SUCCESS` flag file is write under table location, it should be write under partition's location.

### Why are the changes needed?
leave `_SUCCESS` file under table path may cause auth problem. Also cause some jobs using `_SUCCESS` to check write partition finished  hard to work.

### Does this PR introduce _any_ user-facing change?
Static partition insert write `_SUCCESS` under partition location

### How was this patch tested?
Added UT

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala [hunk 1]
```diff
@@ -105,10 +105,22 @@ case class InsertIntoHadoopFsRelationCommand(
...
     val jobId = java.util.UUID.randomUUID().toString
+    // For dynamic partition overwrite, FileOutputCommitter's output path is staging path, files
+    // will be renamed from staging path to final output path during commit job
+    val committerOutputPath = if (dynamicPartitionOverwrite) {
+      FileCommitProtocol.getStagingDir(outputPath.toString, jobId)
+        .makeQualified(fs.getUri, fs.getWorkingDirectory)
+    } else if (staticPartitions.size == partitionColumns.length) {
+      val defaultLocation = qualifiedOutputPath.suffix(
+        "/" + PartitioningUtils.getPathFragment(staticPartitions, partitionColumns)).toString
+      customPartitionLocations.getOrElse(staticPartitions, defaultLocation)
```
Comment (AngersZhuuuu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala:116]: Here will change the value of committerOutputPath in static partition insert to partition location. It's reasonable.
This value is only used in 
 1.    FileOutputFormat.setOutputPath(job, new Path(outputSpec.outputPath))
 2.   Assigned to WriteJobDescription's path parameter, it's not used at all

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

File: core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala [hunk 2]
```diff
@@ -204,7 +204,8 @@ object FileCommitProtocol extends Logging {
       className: String,
       jobId: String,
       outputPath: String,
-      dynamicPartitionOverwrite: Boolean = false): FileCommitProtocol = {
+      dynamicPartitionOverwrite: Boolean = false,
+      staticPartitionInsert: Boolean = false): FileCommitProtocol = {
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala:208]: hmm, adding one more parameter to the commit protocol API looks a bit overkill to fix this problem, which users can just disable writing the `_SUCCESS` file.

Comment (AngersZhuuuu) [reply core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala]: > hmm, adding one more parameter to the commit protocol API looks a bit overkill to fix this problem, which users can just disable writing the `_SUCCESS` file.

Update to use flag set in `hadoopConf`

**Pull request closed without merging**